### ISSUE TITLE
Traceback with 0.3 (latest from PyPI)

### ISSUE THREAD

**Issue by msabramo** [Kotti/kotti_navigation#12]
I pip installed kotti_navigation and added it to `kotti.configurators` but didn't configure where to show the nav yet. When I view a page, I get:

```pytb
  File "/Users/marca/python/virtualenvs/kotti_inventorysvc/lib/python2.7/site-packages/chameleon/py26.py", line 5, in lookup_attr
    return getattr(obj, key)
  File "/Users/marca/dev/git-repos/Kotti/kotti/views/util.py", line 101, in __getattr__
    for snippet in objectevent_listeners(event):
  File "/Users/marca/dev/git-repos/Kotti/kotti/events.py", line 169, in __call__
    results.append(handler(event))
  File "/Users/marca/dev/git-repos/Kotti/kotti/views/slots.py", line 105, in <lambda>
    lambda ev: _render_view_on_slot_event(view_name, ev, params))
  File "/Users/marca/dev/git-repos/Kotti/kotti/views/slots.py", line 79, in _render_view_on_slot_event
    result = render_view(context, view_request, view_name)
  File "/Users/marca/dev/git-repos/pyramid/pyramid/view.py", line 114, in render_view
    iterable = render_view_to_iterable(context, request, name, secure)
  File "/Users/marca/dev/git-repos/pyramid/pyramid/view.py", line 87, in render_view_to_iterable
    response = render_view_to_response(context, request, name, secure)
  File "/Users/marca/dev/git-repos/pyramid/pyramid/view.py", line 60, in render_view_to_response
    return view(context, request)
  File "/Users/marca/dev/git-repos/pyramid/pyramid/config/views.py", line 352, in rendered_view
    result = view(context, request)
  File "/Users/marca/python/virtualenvs/kotti_inventorysvc/lib/python2.7/site-packages/kotti_navigation/views.py", line 125, in navigation_widget
    items = get_children(root, request)
  File "/Users/marca/python/virtualenvs/kotti_inventorysvc/lib/python2.7/site-packages/kotti_navigation/views.py", line 26, in get_children
    if c.in_navigation and c.__class__ not in ex_cts]
TypeError: 'in <string>' requires string as left operand, not DeclarativeMeta

 - Expression: "page_slots.left"
 - Filename:   ... marca/dev/git-repos/Kotti/kotti/templates/view/master.pt
 - Location:   (line 40: col 28)
 - Source:     tal:condition="page_slots.left"
                              ^^^^^^^^^^^^^^^
 - Arguments:  repeat: {...} (0)
               renderer_name: kotti:templates/view/document.pt
               req: <Request - at 0x1090ce190>
               request: <Request - at 0x1090ce190>
               renderer_info: <RendererHelper - at 0x107a65a10>
               api: <TemplateAPI - at 0x1091024d0>
               context: <Host my_host at 0x1090e2ad0>
               view: <function view at 0x106cd07d0>
```

**Comment by msabramo** [comment]
This was with kotti from git master and kotti_navigation from PyPI, which is probably not a good combination.

But when I tried to use kotti_navigation from git master, I ran into another problem: https://github.com/Kotti/kotti_navigation/issues/13

**Comment by davidemoro** [comment]
Confirmed, same issue (page_slots.left).

**Comment by davidemoro** [comment]
HI @msabramo,

the master has 21 tests failures. I fixed the page_slots.left issue on https://github.com/truelab/kotti_navigation (now only 6 failures, 15 broken tests solved). Now I'm able to display my navigation left portlet, hope it might help.

PS: I tried to play with the kotti_navigation.navigation_widget.include_content_types and kotti_navigation.navigation_widget.exclude_content_types options but I'm not able to see any change (even using kotti.resources.Image or just Image). Any hints?

**Comment by j23d** [comment]
I suspect that 0.3 is not working with the latest 1.0.0-alpha from Kotti, so Kotti in an older version and the appropriate requirements have to be used in order to get this working. I have to check this and will update the 0.3 branch and the latest stable on pypi.

The most current version, however it is also outdated, is in the branch ``switch_to_kotti_settings``, see 
https://github.com/Kotti/kotti_navigation/tree/switch_to_kotti_settings. This needs some more work, an update to be compatible with Kotti 1.0 and new docs. I will check this and likely merge it into the master.

**Comment by j23d** [comment]
@davidemoro: please see https://github.com/Kotti/kotti_navigation/tree/0.3.x#excluding-content-types again. If you set e.g. kotti.resources.Document as content types to exclude in your ini, the default documents will not appear in the navigation anymore. This feature was mostly useful for me when I have own content types that should not appear in the navigation.

**Comment by j23d** [comment]
@msabramo: I try again to reproduce this error in a fresh virtualenv, but failed to do so. The navigation starts with some defaults in the left slot with the fresh released 0.3.1 version from pypi (https://pypi.python.org/pypi/kotti_navigation/0.3.1) and also not with the current master. I close the ticket. If the problem persist on your side, please reopen it.

**Comment by davidemoro** [comment]
@j23d: confirmed, all fine with the fresh released 0.3.1 version.
